src/server_manager/electron_app/preload.ts

```diff
@@ -47,9 +47,16 @@ if (sentryDsn) {
   });
 }
 
-contextBridge.exposeInMainWorld('trustCertificate', (host: string, fingerprint: string) => {
-  return ipcRenderer.sendSync('trust-certificate', host, fingerprint);
-});
+contextBridge.exposeInMainWorld(
+  'fetchWithPin',
```

Comment:
Define this function in a renderer code library and depend on it directly. No need for this on preload.

Reply:
I tried this, but importing `ipcRenderer` in the renderer code breaks `npm run action server_manager/web_app/test`, which runs that code in a non-Electron Karma context.

Instead, I was able to find a way to import the `HttpRequest` and `HttpResponse` types into `preload.d.ts`.